cli/src/pcluster/commands.py

```diff
@@ -520,6 +520,14 @@ def delete(args):
                     ).ljust(80)
                     sys.stdout.write("\r%s" % resource_status)
                     sys.stdout.flush()
+                elif isinstance(result, ApiFailure):
+                    # If stack is already deleted
+                    if f"Cluster {args.cluster_name} doesn't exist." in result.message:
+                        LOGGER.warning(f"\nCluster {args.cluster_name} has already been deleted or does not exist.")
```

Comment:
Why treating `doesn't exist` error differently?

Reply:
Is this path reachable? I guess all `doesn't exist` exception goes to line 542?

Reply:
Because `result = PclusterApi().describe_cluster(cluster_name=args.cluster_name, region=utils.get_region())` in while loop will get APIFailure at the end of deletion, the stack has been deleted. It's not the ClientError, so will go to this branch.